visivo/models/base/parent_model.py

```diff
@@ -18,39 +19,45 @@ def child_items(self):
     def dag(self, node_permit_list=None):
         dag = nx.DiGraph()
         dag.add_node(self)
-        self.traverse_fields(items=self.child_items(), 
-                             parent_item=self, 
-                             dag=dag, 
-                             node_permit_list=node_permit_list, 
-                             root=self)
+        self.traverse_fields(
+            items=self.child_items(),
+            parent_item=self,
+            dag=dag,
+            node_permit_list=node_permit_list,
+            root=self,
+        )
         return dag
 
     def traverse_fields(self, items: List, parent_item, dag, node_permit_list, root):
         for item in items:
-            if node_permit_list is None or item in node_permit_list: 
+            if node_permit_list is None or item in node_permit_list:
                 dereferenced_item = item
                 if BaseModel.is_ref(item):
                     name = NamedModel.get_name(obj=item)
-                    dereferenced_items = ParentModel.all_descendants_with_name(
-                        name=name, dag=dag, from_node=root
+                    dereferenced_item = self._get_dereferenced_item(
+                        name=name,
+                        dag=dag,
+                        root=root,
+                        item=item,
+                        parent_item=parent_item,
+                    )
+                elif isinstance(item, DefaultTarget):
+                    name = root.defaults.target_name
+                    dereferenced_item = self._get_dereferenced_item(
+                        name=name,
+                        dag=dag,
+                        root=root,
+                        item=item,
+                        parent_item=parent_item,
                     )
-                    if len(dereferenced_items) == 1:
-                        dereferenced_item = dereferenced_items[0]
-                    else:
-                        raise PydanticCustomError(
-                            "bad_reference",
-                            f'The reference "{item}" on item "{parent_item.id()}" does not point to an object.',
```

Comment:
Did you find that the default error produced a better message?

Reply:
No I just moved this into a private function down the page.